src/common/applyLanguage.ts

```diff
@@ -4,13 +4,39 @@ import { Language, defaultDetokenizer, Plugin } from "./Language";
 import { programToSpine } from "./Spine";
 import polygolfLanguage from "../languages/polygolf";
 
+export type OptimisationLevel = "none" | "heuristic" | "full";
+export type Objective = "bytes" | "chars";
+export interface SearchOptions {
+  level: OptimisationLevel;
+  objective: Objective;
+  objectiveFunction: (x: string) => number;
+}
+
+export function searchOptions(
+  level: OptimisationLevel,
+  objective: Objective,
+  objectiveFunction?: (x: string) => number
+): SearchOptions {
+  return {
+    level,
+    objective,
+    objectiveFunction:
+      objectiveFunction ??
+      (objective === "bytes"
+        ? (x) => Buffer.byteLength(x, "utf-8")
+        : (x) => x.length),
```

Comment:
Note `"😊".length` gives 2, but we expect 1. Try copying [charLen](https://github.com/code-golf/code-golf/blob/13733cfd472011217031fb9e733ae9ac177b234b/js/_util.ts#L7) from code-golf/code-golf.